lib/models/requestResponse.js

```diff
@@ -177,10 +181,7 @@ class RequestResponse {
       case 'referer':
       case 'retry-after':
       case 'user-agent':
-        Object.keys(this[_headers])
-          .filter(key => key.toLowerCase() === name.toLowerCase())
-          .forEach(key => delete this[_headers][key]);
-        this[_headers][name] = value;
+        this[_headers][lowercased] = value;
```

Comment:
nitpicking: Why force a lower cased value for the user agent?
Common practices are to keep it camel-cased (Chrome sends `User-Agent`). 

This is really non-blocking as this header does not make much sense in a response though.